activerecord/test/cases/associations/required_test.rb

```diff
@@ -22,30 +22,6 @@ class Child < ActiveRecord::Base
     @connection.drop_table "children", if_exists: true
   end
 
-  test "belongs_to associations are not required by default" do
```

Comment:
That is a good question.... does that test really test something?

Because reading this,  https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/builder/belongs_to.rb#L127 I assumed that if defined a `required: true` it would be replaced for `optional: false`

If not, apparently it would default to https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/builder/belongs_to.rb#L131